### PR TITLE
zfp: 0.5.5 -> 1.0.0

### PR DESCRIPTION
**Pull request by AtilaSaraiva**
###### Description of changes

This release is not ABI compatible with prior releases due to numerous changes
to function signatures and data structures like zfp_field. However, few of
the API changes, other than to the cfp C API for compressed arrays, should
impact existing code. Note that numerous header files have been renamed or
moved relative to prior versions.

https://github.com/LLNL/zfp/releases/tag/1.0.0

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by AtilaSaraiva** [comment]
Result of `nixpkgs-review pr 188847` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>zfp</li>
  </ul>
</details>

File: pkgs/tools/compression/zfp/default.nix [hunk 1]
```diff
@@ -52,7 +52,7 @@ stdenv.mkDerivation rec {
     homepage = "https://computing.llnl.gov/projects/zfp";
     description = "Library for random-access compression of floating-point arrays";
     license = licenses.bsd3;
-    maintainers = [ maintainers.spease ];
+    maintainers = with maintainers; [ spease atila ];
     # 64-bit only
     platforms = platforms.aarch64 ++ platforms.x86_64;
```
Comment (SuperSandro2000) [inline pkgs/tools/compression/zfp/default.nix:57]:
```suggestion
    platforms = platforms.aarch64 ++ platforms.x86_64;
    # error: '_Float16' was not declared in this scope; did you mean '_Float64'?
    broken = stdenv.isAarch64;
```

**Comment by risicle** [comment]
On macos 10.15:

```
/tmp/nix-build-zfp-1.0.0.drv-0/source/examples/diffusion.cpp:31:61: error: _Float16 is not supported on this target
  typedef zfp::array2< double, zfp::codec::generic2<double, _Float16> > array2h;
                                                            ^
```

**Comment by spease** [comment]
LGTM

**Comment by AtilaSaraiva** [comment]
Let's see if now it works

File: pkgs/tools/compression/zfp/default.nix (hunk 1 above)
Comment (AtilaSaraiva) [reply pkgs/tools/compression/zfp/default.nix:57]: Thanks for the review! Fixed! But darwin_x86 is still failling.

Comment (SuperSandro2000) [reply pkgs/tools/compression/zfp/default.nix:57]: I would suggest to mark darwin broken.

**Comment by SuperSandro2000** [comment]
x86_64-darwin still fails to build. It should be marked broken.

**Comment by spease** [comment]
There’s another PR to upgrade it: #199892

**Comment by AtilaSaraiva** [comment]
Perfect then, thx.

**Pull request closed without merging**